### ISSUE TITLE
Some mp3 files not getting detected

### ISSUE THREAD

**Issue by sumeetphadnis** [mscdex/mmmagic#35]
I am trying to detect the mime type of audio file uploaded to my server running Ubuntu, node 0.10.33. For some valid mp3 files the mime type is not detected properly and application/octet-stream is returned.

I have attached a screenshot of the terminal window:
![screen shot 2015-01-01 at 6 18 46 pm](https://cloud.githubusercontent.com/assets/2618285/5592193/483b1b88-91e3-11e4-9359-962905897387.png)

**Comment by mscdex** [comment]
Do you have a copy of the mp3 I can test with?

**Comment by sumeetphadnis** [comment]
Thanks for the quick revert! [Here is a file](https://www.dropbox.com/s/86yl7vcinm9bvwh/46847-1ftiagy.mp3?dl=0) you can test with. 

Happy new year!

**Comment by mscdex** [comment]
Ok, I see what the problem is now. The ID3 tag length is incorrect in that file. That mp3 file has a stored ID3 tag length of 4086 bytes, whereas the actual length is 5130 bytes. So when libmagic jumps to 4086 (4096 including the ID3 header) bytes in to look for the actual audio (mp3) header, it doesn't see the correct data so it never matches.

I'm not sure there is anything libmagic can do for corrupt files like this. One workaround might be to check the file description and if it detected it as an audio file with ID3 but the mime type is still application/octet-stream, use a utility like `id3v2` to rewrite the id3v2 data. I did this via `id3v2 --id3v2-only file.mp3`, which only rewrites the id3v2 tag (leaves the id3v1 tag alone if there is one). After doing that, the length was corrected so the mime type was given correctly (and the description actually showed details on the audio format inside the file).
